src/staking/staking_pool.h

```diff
@@ -0,0 +1,22 @@
+#ifndef ELECTRIC_CASH_STAKING_POOL_H
+#define ELECTRIC_CASH_STAKING_POOL_H
+#include "amount.h"
+
+class StakingPool {
+
+private:
+    static StakingPool* srpInstance;
+    CAmount balance = 0;
+    StakingPool(){};
+
+public:
+    StakingPool(StakingPool& copy) = delete;
```

Comment:
Singleton classes also shouldn't be be assignable
`void operator=(const StakingPool&) = delete;`

Reply:
done